pandas/core/construction.py

```diff
@@ -762,6 +764,9 @@ def _try_cast(
 
         subarr = maybe_cast_to_integer_array(arr, dtype)
     else:
-        subarr = np.array(arr, dtype=dtype, copy=copy)
+        if using_copy_on_write():
+            subarr = np.array(arr, dtype=dtype, copy=copy, order="F")
```

Comment:
Why is the order="F" specifically needed for CoW? (and can you add a comment about it)

Reply:
This is about #50756

Reply:
Yes, I understand that it's related to that, but I don't understand why we would only do it for CoW? The default is not to copy arrays (without CoW) at the moment, which creates this inefficient layout; but so if the user manually specifies copy=True in the constructor, why not directly copy it to the desired layout in all cases without the if/else check for CoW?

Reply:
Ah did not think about this, will add

Reply:
xref #44871 took a look at preserving order when doing copy.  there are some tradeoffs here

Reply:
Based on @jbrockmendel s comment above I think we should leave it out for now

Reply:
> Based on @jbrockmendel s comment above I think we should leave it out for now

Leave it out in general (from this PR), or you mean not do it for the default mode for now?

Reply:
Definitely default mode and maybe also split the copy and the layout change into two PRs?

Reply:
reiterating preference for this to be separate.  The two of you are much more familiar with the CoW logic than most of the rest of us; i get antsy when i see using_copy_on_write checks appearing in new places

Reply:
I removed all order relevant changes, is more straightforward now